### PR TITLE
Fixed #26157 #25321 -- Added sql/params to extra context of schema logger.

### PR DESCRIPTION
**Pull request by MarkusH**
https://code.djangoproject.com/ticket/26157
https://code.djangoproject.com/ticket/25321

Supersedes #6256

### CODE REVIEW

**Comment by akki** [comment]
Thanks @MarkusH for completing this.

File: tests/logging_tests/tests.py [hunk 1]
```diff
@@ -475,3 +476,23 @@ def test_custom_logging(self):
         out, err = self.run_manage(['check'])
         self.assertNoOutput(err)
         self.assertOutput(out, "System check identified no issues (0 silenced).")
+
+
+class SchemaLoggerTests(SimpleTestCase):
```
Comment (timgraham) [inline tests/logging_tests/tests.py]: I'm more inclined to put this in `tests/schema/test_logging.py`.

File: docs/topics/logging.txt [hunk 2]
```diff
@@ -584,6 +584,12 @@ Logs the SQL queries that are executed during schema changes to the database by
 the :doc:`migrations framework </topics/migrations>`. Note that it won't log the
 queries executed by :class:`~django.db.migrations.operations.RunPython`.
 
+.. versionadded:: 1.10
```
Comment (timgraham) [inline docs/topics/logging.txt]: Putting these markers in the middle of several paragraphs can be ambiguous. I prefer if you put it at the bottom of the section with a note like "The extra context was added."

File: docs/topics/logging.txt [hunk 3]
```diff
@@ -584,6 +584,12 @@ Logs the SQL queries that are executed during schema changes to the database by
...
+
+Messages to this logger have ``params`` and ``sql`` but not ``duration`` in
```
Comment (timgraham) [inline docs/topics/logging.txt]: rephrase: "params and sql in their extra context (but unlike django.db.backends, not duration)".

File: django/test/utils.py [hunk 4]
```diff
@@ -510,7 +510,10 @@ def patch_logger(logger_name, log_level):
     calls = []
 
     def replacement(msg, *args, **kwargs):
-        calls.append(msg % args)
+        if log_kwargs:
```
Comment (timgraham) [inline django/test/utils.py]: DRYer?
```
call = msg % args
calls.append((call, kwargs) if log_kwargs else call)
```

**Comment by timgraham** [comment]
Suggested a more descriptive commit message in the PR title.

File: tests/logging_tests/tests.py (hunk 1 above)
Comment (MarkusH) [inline tests/logging_tests/tests.py]: All the other loggers and logging features (as far as being tested) are handled n `tests.logging_tests` which is why I'm more inclined to keep it there.

**Comment by MarkusH** [comment]
Merged in 1cb65b8a77bd1b15f094b5857468c107784aad90

**Pull request closed without merging**